docs/apis/calcite.md

```diff
@@ -0,0 +1,235 @@
+                     FROM ORDERS 
+                     WHERE ID > 3 AND PRODUCT LIKE 'paint%'";
+   sqlEnv.executeSQL(dag, sql);
+ }// populateDAG finished
+ 
+ public static String apex_concat_str(String s1, String s2)
+ {
+      return s1 + s2;
+ } 
+}
+```      
+### Constructing SQLExecEnvironment
+Class _**SQLExecEnvironment**_ provides a starting point and a simple way to define metadata needed for running SQL statement. SQLExecEnvironment.getEnvironment() creates a new instance of SQLExecEnvironment.  
+
+```java
+// Creates SQLExecEnvironment instance by using static method getEnvironment
+SQLExecEnvironment sqlEnv = SQLExecEnvironment.getEnvironment();
+```
+
+### Registering tables with SQLExecEnvironment
+Next, we need to register tables which can be used in a query. For this purpose, we can use _registerTable_ API from SQLExecEnvironment.
+```java
+// Register KafkaEnpoint as "ORDERS" table with kafka topic and data format as CSV
+sqlEnv = sqlEnv.registerTable( 
+                              "ORDERS", 
+                              new KafkaEndpoint("localhost:9090", 
+                                                "inputTopic", 
+                                                new CSVMessageFormat(<>))
+                            );
+
+// Register FileEndpoint as "SALES" table with file path and data format as csv
+sqlEnv = sqlEnv.registerTable( 
+                              "SALES", 
+                              new FileEndpoint("/tmp/output", 
+                                               "out.file", 
+                                               new CSVMessageFormat(<>))
+                            );
+```
+
+**_"registerTable"_** API takes the name of the table and an instance of endpoint as parameters. Endpoint signifies data storage mechanism and type of source/destination for the data. Each of these endpoints requires a different type of configurations and possibly data formats. In this example, data format is defined using _MessageFormat_ by specifying it at the endpoint creation. _CSVMessageFormat_ can be configured with schema string as follows:
+```sql
+{
+  "separator": ",",
+  "quoteChar": "\"",
+  "fields": [
+    {
+      "name": "RowTime",
+      "type": "Date",
+      "constraints": {
+        "format": "dd/MM/yyyy hh:mm:ss Z"
+      }
+    },
+    {
+      "name": "id",
+      "type": "Integer"
+    },
+    {
+      "name": "Product",
+      "type": "String"
+    },
+    {
+      "name": "units",
+      "type": "Integer"
+    }
+  ]
+}
+```
+Schema string is a JSON string defining a separator character for schema, quote character for fields with String types and fields along with their name, type and additional constraints if there are any.
+
+Following data endpoints are supported: 
+
+* **KafkaEnpoint**
+: To define Kafka Endpoint we need to specify Kafka broker ( host:port ), topic name and MessageFormat as seen in line 1 in above code.
```

Comment:
"Kafka Endpoint" => a Kafka endpoint
Kafka broker ( host:port ) => the Kafka broker (as host:port)
in above code => in the code above